docs/developer/architecture/logs.rst

```diff
@@ -0,0 +1,215 @@
+Log Centralization
+==================
+
+Context
+-------
+
+MetalK8s value is to provide, out of the box, some services in order to ease
+the monitoring and operation of the platform as well as workloads running on
+top of it.
+
+It currently provides monitoring service, powered by Prometheus and
+AlertManager, in order to expose metrics and alerts for both control-plane and
+workload-plane components as well as for cluster nodes HW and OS.
+
+The logs generated by the platform and the workloads constitute an essential
+piece of information when it comes to understanding the root cause of a failure
+or a performance degradation. Because of the distributed nature of MetalK8s and
+workloads running on top of it, the administrators need some tooling to ease
+the analysis of near real time and past logs, from a central endpoint. As such
+these logs should be stored on the platform, for a configurable period.
+Browsing the logs is accessible through an API and a UI. The UI should ease
+correlation between logs and health/performance KPIs as well as alerts.
+
+For organisations having their own Log centralization system (like Splunk or
+Elasticsearch), MetalK8s should provide some documentation in order guide the
+customer to deploy and configure its own log collection agent.
+
+The following requirements focus on application logs. Audit logs are not part
+of the requirements.
+
+Requirements
```

Comment:
I would add a requirement on getting the log "visible". IE : the difference between the time when the logs is generated and "printed" (visible in kubectl logs) and the time when it appears in the UI.
Also, I would precise if loss of logs is acceptable and in which circumstances it is and how (if ?) the end user is notified of the loss.